larray/core/array.py

```diff
@@ -6019,7 +6019,9 @@ def to_hdf(self, filepath, key, *args, **kwargs):
         >>> a.to_hdf('test.h5', 'a')  # doctest: +SKIP
         """
         key = _translate_key_hdf(key)
-        self.to_frame().to_hdf(filepath, key, *args, **kwargs)
+        with LHDFStore(filepath) as store:
+            store.put(key, self.to_frame())
+            store.get_storer(key).attrs.type = 'Array'
```

Comment:
Also, it somewhat bothers me to have to convert to a pandas object first. I know the LHDFStore is internal for now, but when we make it part of the API, this will need to be changed. We should be able to write something like:

```python
arr = ndtest((2, 3))
with HDFStore(filepath) as store:
    store.put('arr', arr)
```

Reply:
For the record: see #614